test/fake/fake-currency-plugin.js

```diff
@@ -56,12 +59,22 @@ const nop: Function = () => {}
 
 type State = {
   balance: number,
+  stakedBalance: number,
   tokenBalance: number,
   blockHeight: number,
   progress: number,
   txs: { [txid: string]: EdgeTransaction }
 }
 
+const asState = asObject({
+  balance: asOptional(asNumber),
+  stakedBalance: asOptional(asNumber),
+  tokenBalance: asOptional(asNumber),
+  blockHeight: asOptional(asNumber),
+  progress: asOptional(asNumber),
+  txs: asOptional(asMap((raw: any) => raw))
+})
```

Comment:
Shouldn't this be `asMaybe`? You were checking `typeof x === 'number'` before, now you're checking if it's not null/undefined. Not sure if it matters here in the context of the mocks, but if `changeUserSettings` receives an argument where one of the properties is defined but doesn't pass the cleaner, then there will be an exception thrown by the cleaner. Using `asMaybe` would avoid this, but this may not need to be avoided if we know that this case doesn't exist.

Reply:
That was the intent. The `changeUserSettings` method is being called make the test plugin do things. If somebody passes the wrong data type, the plugin should complain loudly, rather than silently "do nothing", which could really mess with the test.